R/mean_var.R

```diff
@@ -142,6 +142,7 @@ setMethod(
 
     n <- sum(w)
     p <- tapply(w, x, function(r) if(n == 0) 0 else sum(r)/n)
+    p[is.na(p)] <- 0 #for empty level in factor, converts proportion from NA to 0.
```

Comment:
Did you try the following?

```r
tapply(w, x, function(r) if(n == 0) 0 else sum(r)/n, default = 0)
```

This worked in a couple of test cases for me..

Reply:
Good call on the `default` argument! That is tidier, I've revised the code and added myself as a contributor.